app/api/entities/reaction_variation_entity.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module Entities
+  class ReactionVariationEntity < ApplicationEntity
+    expose(
+      :id,
+      :properties,
+      :startingMaterials,
+      :reactants,
+      :products,
+    )
+
+    def properties
+      {}.tap do |properties|
+        properties[:temperature] = ReactionVariationPropertyEntity.represent(object[:properties][:temperature])
+        properties[:duration] =  ReactionVariationPropertyEntity.represent(object[:properties][:duration])
+      end
+    end
+
+    def materials(material_type)
+      {}.tap do |materials|
+        object[material_type].each do |k, v|
+          materials[k] = ReactionVariationMaterialEntity.represent(v)
+        end
+      end
+    end
+
+    def startingMaterials
+      materials(:startingMaterials)
+    end
+
+    def reactants
+      materials(:reactants)
+    end
+
+    def products
+      materials(:products)
+    end
+  end
+
+  class ReactionVariationPropertyEntity < ApplicationEntity
+    expose(
+      :value,
+      :unit,
+    )
+  end
+
+  class ReactionVariationMaterialEntity < ApplicationEntity
+    expose(
+      :value,
+      :unit,
+      :aux,
+    )
+
+    def aux
+      {}.tap do |aux|
+        aux[:coefficient] = object[:aux][:coefficient]
```

Comment:
Wenn aux nur exakt die Keys enthält die du hier abfragst kannst du auch einfach object[:aux] zurückgeben und sparst dir die Rubocop Warnung. Ansonsten ist mir die Datenstruktur n bissl unklar, ob man es nicht hinbekommt in der ReactionVariationMaterialEntity aux als nested Key zu implementieren. Das müsste er bei nem Hash eigentlich auflösen können (musst evtl guggn bzgl String/Hash Keys aber ansonsten passt das)

Reply:
Eine [nested exposure](https://github.com/ruby-grape/grape-entity#nested-exposure) funktioniert eigenartigerweise nicht:
```
expose :aux do
  expose(
	:coefficient,
	:isReference,
	:referenceAmount,
	:loading,
	:purity,
	:molarity,
        :molecularWeight,
	:sumFormula,
	:yield,
  )
```
Ich habe es jetzt mit einer Sub-Entity geloest, das ist immerhin ideomatischer und etwas besser zu lesen (b97ed98).